objects/o_solid_oneway/Create_0.gml

```diff
@@ -0,0 +1,114 @@
+#region Initialize
+	
+	//movement variables
+	xspd = 0;
+	xspd_remainder = 0;
+
+	yspd = 0;
+	yspd_remainder = 0;
+
+	dir = 1; //movement direction
+	
+#endregion
+
+#region Utils
+
+	//populate a list of all actors who are riding the solid
+	//Returns: a DS list of actors riding the solid
+	function get_rider_list() {
+		var _list_of_riders = ds_list_create();
+		ds_list_clear(_list_of_riders);
+
+		with (o_actor) {
+			if (is_riding(other)) ds_list_add(_list_of_riders,id);
+		}
+		
+		return _list_of_riders;
+	}
+
+#endregion
+
+#region Movement and collision
+
+	//Note: movement for solids should normally be executed in the Begin Step event
+	//pixel by pixel movement and collision checks, carrying fractional remainders to next frame
+	//first try and push any actors (if an overlap between an actor and moving solid occurs, move actor to the appropriate leading edge)
+	//then carry any riding actors
+	//Param: _xspd,_yspd (amount to move solid in current frame)
+	//Param: _collision_event (function to execute when a collision is detected (defaults to no action))
+	function move_x(_xspd, collision_event = function() {}) {
```

Comment:
Potential idea for another ticket if we're going to have more move_x and move_y functions in other parents. Might want to refactor the move_x and move_y functions into the script too. Then the parent objects can have that `move_and_collide(x, y, _x_coll_func, _y_coll_func)` function which can call the appropriate move_x and move_y